server/src/main/java/com/cloud/resource/ResourceManagerImpl.java

```diff
@@ -687,6 +688,10 @@ public Discoverer getMatchingDiscover(final Hypervisor.HypervisorType hypervisor
         if ((clusterName != null || clusterId != null) && podId == null) {
             throw new InvalidParameterValueException("Can't specify cluster without specifying the pod");
         }
+        if (!hypervisorType.equalsIgnoreCase(HypervisorType.VMware.toString()) &&
+                (Strings.isNullOrEmpty(username) || Strings.isNullOrEmpty(password))) {
+            throw new InvalidParameterValueException("Username and Password need to be provided.");
+        }
```

Comment:
maybe excessive but also a tat safer:
```suggestion
        if (!HypervisorType.VMware.toString().equalsIgnoreCase(hypervisorType) &&
                (Strings.isNullOrEmpty(username) || Strings.isNullOrEmpty(password))) {
            throw new InvalidParameterValueException("Username and Password need to be provided.");
        }
```